src/toolkits/style_transfer/style_transfer.hpp

```diff
@@ -119,6 +119,33 @@ class EXPORT style_transfer : public ml_model_base {
     return variant_get_value<T>(get_state().at(key));
   }
 
+  template <typename T>
+  typename std::map<std::string, T>::iterator _read_iter_opts(
+      std::map<std::string, T>& opts, const std::string& key) const {
+    auto iter = opts.find(key);
+    if (iter == opts.end())
+      log_and_throw("Expected option \"" + key + "\" not found.");
+    return iter;
+  }
+
+  template <typename T>
+  T read_opts(std::map<std::string, turi::variant_type>& opts,
+              const std::string& key, bool remove_item = false) const {
+    auto iter = _read_iter_opts<turi::variant_type>(opts, key);
+    T value = variant_get_value<T>(iter->second);
+    if (remove_item) opts.erase(iter);
+    return value;
+  }
+
+  template <typename T>
+  T read_opts(std::map<std::string, turi::flexible_type>& opts,
+              const std::string& key, bool remove_item = false) const {
+    auto iter = _read_iter_opts<turi::flexible_type>(opts, key);
+    T value = iter->second.get<T>();
+    if (remove_item) opts.erase(iter);
+    return value;
+  }
```

Comment:
Can we remove the `remove_item` argument? It seems incompatible with these being `const` methods anyway